drivers/flash/flash_uwp.c

```diff
@@ -75,13 +75,17 @@ static int flash_uwp_read(struct device *dev, off_t offset, void *data,
 {
 	int ret = 0;
 
+	struct flash_uwp_config *cfg = DEV_CFG(dev);
+	struct spi_flash *flash = &(cfg->flash);
+
 	if (!len) {
 		return 0;
 	}
 
 	flash_uwp_lock(dev);
 
-	memcpy(data, (void *)((u32_t)CONFIG_FLASH_BASE_ADDRESS + offset), len);
+	ret = !(flash->read(flash, ((u32_t)CONFIG_FLASH_BASE_ADDRESS + offset),
```

Comment:
Interfaces in Zephyr usual use return 0 to be success, can you remove "!" to keep it consistent with others.

Reply:
fixed